### DISCUSSION TITLE
Renaming a module in OCaml is broken

### DISCUSSION THREAD

**Discussion by theteachr** [helix-editor/helix#11696] [Troubleshooting]
### Summary

Renaming a module in OCaml goofs up not only the text around, but also undo and redo actions.

### Reproduction Steps

 

https://github.com/user-attachments/assets/2d5c5cbe-6442-4ba4-81d3-786c8738e89a

```ocaml
(* a.ml *)

type a
type b

module Event = struct
  type _ t = A : a t | B : b t
end

open Event

let process : a t -> unit = function
  | A  -> ()
```

- `hx a.ml`
- Put the cursor anywhere on `Event`
- <kbd>Space</kbd> <kbd>R</kbd>
- <kbd>Shift</kbd> + <kbd>A</kbd> <kbd>Enter</kbd>
- <kbd>U</kbd>
- <kbd>Shift</kbd> + <kbd>U</kbd>

### Platform

macOS

### Terminal Emulator

iTerm

### Installation Method

source

### Helix Version

helix 24.3 (0c8d51ee)

**Comment by the-mikedavis** [comment]
I can't reproduce this using the latest Helix master or 24.07 or 24.03 and the latest ocamllsp. I suspect this was a bug in ocamllsp with the textedits it sent for this rename.

**Reply by theteachr** [reply]
I can confirm that this is no longer an issue with the latest (1.19.0) `ocaml-lsp`. But I can't use the latest one for my current switch, that's using 1.17.0 of `ocaml-lsp`. But I also understand that this is not an issue with Helix alone.
